### PR TITLE
📖 Update simple-cluster example to v1alpha2

### PR DESCRIPTION
**Pull request by erwinvaneyk**
Currently there is still a v1alpha1 example in the current repo. To avoid confusing new users, and have a simple example to share, I updated the example to v1alpha2.

The structure of the example remains the same: deploy 1 cluster with 1 control plane node and 1 worker node.

I verified the example locally (on Mac), using the `gcr.io/kubernetes1-226021/manager:dev` image.

### CODE REVIEW

File: examples/simple-cluster.yaml [hunk 1]
```diff
@@ -1,38 +1,108 @@
...
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+kind: KubeadmConfig
+metadata:
+  name: worker-0-config
+  namespace: default
 spec:
-  providerSpec: {}
-  versions:
-    kubelet: "v1.14.2"
+  joinConfiguration:
```
Comment (detiber) [inline examples/simple-cluster.yaml]:
```suggestion
  joinConfiguration: {}
```

File: examples/simple-cluster.yaml [hunk 2]
```diff
@@ -1,38 +1,108 @@
...
+    discovery: {}
```
Comment (detiber) [inline examples/simple-cluster.yaml]: with the above change you can remove `discovery` and `nodeRegistration`

File: examples/simple-cluster.yaml [hunk 3]
```diff
@@ -1,38 +1,108 @@
...
+    namespace: default
 ---
-apiVersion: "cluster.k8s.io/v1alpha1"
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+kind: KubeadmConfig
+metadata:
+  name: controlplane-0-config
+  namespace: default
+spec:
+  clusterConfiguration:
```
Comment (detiber) [inline examples/simple-cluster.yaml]:
```suggestion
  clusterConfiguration: {}
```

Comment (detiber) [reply examples/simple-cluster.yaml]: With the above change you can remove all the configuration under `clusterConfiguration`

File: examples/simple-cluster.yaml [hunk 4]
```diff
@@ -1,38 +1,108 @@
...
+      type: ""
+    etcd: {}
+    imageRepository: ""
+    kubernetesVersion: ""
+    networking:
+      dnsDomain: ""
+      podSubnet: ""
+      serviceSubnet: ""
+    scheduler: {}
+  initConfiguration:
```
Comment (detiber) [inline examples/simple-cluster.yaml]:
```suggestion
  initConfiguration: {}
```

Comment (detiber) [reply examples/simple-cluster.yaml]: With the above change you can remove the other initConfiguration config here.

File: examples/simple-cluster.yaml [hunk 5]
```diff
@@ -1,38 +1,108 @@
...
 metadata:
-  name: my-control-plane
   labels:
-    cluster.k8s.io/cluster-name: my-cluster
-    set: "controlplane"
+    cluster.x-k8s.io/cluster-name: my-cluster
+    cluster.x-k8s.io/control-plane: "true"
+  name: controlplane-0
+  namespace: default
 spec:
```
Comment (detiber) [inline examples/simple-cluster.yaml]: I'd recommend setting `version` here as well.

**Comment by erwinvaneyk** [comment]
@detiber thanks for the quick review! 😁 

Maybe a bit out-of-scope feedback, but it is pretty unclear what fields are optional and what fields are required in the kubeadm (e.g. `initConfiguration: {}` is required, but not the subfields). Maybe I am just missing the reasoning behind it though

File: examples/simple-cluster.yaml (hunk 2 above)
Comment (erwinvaneyk) [reply examples/simple-cluster.yaml]: Seems like nodeRegistration is required by the validating-webhook
```
error: error validating "mvp/workload/cluster-test/simple-cluster.yaml": error validating data: ValidationError(KubeadmConfig.spec.joinConfiguration): missing required field "nodeRegistration" in io.x-k8s.cluster.bootstrap.v1alpha2.KubeadmConfig.spec.joinConfiguration; if you choose to ignore these errors, turn validation off with --validate=false
```

Comment (erwinvaneyk) [reply examples/simple-cluster.yaml]: Seems like nodeRegistration is required by the validating-webhook
```
error: error validating "cluster-test/simple-cluster.yaml": error validating data: ValidationError(KubeadmConfig.spec.joinConfiguration): missing required field "nodeRegistration" in io.x-k8s.cluster.bootstrap.v1alpha2.KubeadmConfig.spec.joinConfiguration; if you choose to ignore these errors, turn validation off with --validate=false
```

**Comment by chuckha** [comment]
What version of the bootstrap provider did you use?

We added some optional flags for some of these fields. I think it's ok to be explicit in the example here though.
Looks like the example just needs a newline at the end of the file and this will pass CI.

**Comment by erwinvaneyk** [comment]
> What version of the bootstrap provider did you use?

Hi @chuckha, I am not sure what I used then, but I just quickly spun up the cluster with the default image in the kustomization template: `gcr.io/k8s-staging-capi-kubeadm/cluster-api-kubeadm-controller:latest`. Should I use a different one to test with?

**Comment by chuckha** [comment]
@erwinvaneyk no, that's a good image to test with

/lgtm

**Pull request merged**